torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1322,8 +1322,7 @@ def make_singular_matrix_batch_base(size, rank):
             p, l, u = torch.lu_unpack(lu, pivs)
             u_diag_abs = u.diagonal(0, -2, -1).abs()
             u_diag_abs_smallest_idxs = torch.topk(u_diag_abs, k=(n - rank), largest=False).indices
-            u.diagonal(0, -2, -1).div_(u_diag_abs_largest)
-            u[..., u_diag_abs_smallest_idxs] = torch.finfo(dtype).eps
+            u.diagonal(0, -2, -1)[..., u_diag_abs_smallest_idxs] = torch.finfo(dtype).eps
```

Comment:
Doesn't it belong to a different stack of det/lu-related PRs?

Reply:
It was breaking here so I put up a fix in this PR. It so happened that the fix diverged. I'll rever to the lu-stack fix.